docs/how-does-version-resolution-work-in-maven-and-gradle.md

```diff
@@ -0,0 +1,124 @@
+# How does version resolution work in Maven and Gradle?
+
+The build system that you choose for building your Java code determines which
+version resolution algorithm is used to choose the versions of your
+dependencies. Unfortunately, the two most popular build systems in open source
+(Maven and Gradle) use different version resolution algorithms. They also use
+different names for the concept: Maven calls the process *dependency mediation*,
+while Gradle calls the process *version conflict resolution*. In this article,
+we will use the term *version resolution*, omitting the word "conflict" so that
+the process it not confused with program-breaking conflicts that we discuss
+elsewhere. No matter which build system you choose, you are likely to have
+direct or indirect dependencies that use the other one, so you need to
+understand how both work. Note: Our focus is on build systems based on the Maven
+repository system, so we ignore other build environments like Eclipse and Ant
+that work differently.
+
+Before we look at Maven and Gradle specifically, we need to understand what
+happens in the version resolution process. The input of this process is a
```

Comment:
But there is no version resolution process without Maven or Gradle. You can describe what exists in the repository, and you can describe what Maven and Gradle do with the information they find there. Nothing else exists.

Reply:
I will rework this to refer to "the concepts of version resolution in general" - there are logical steps that any build system needs to perform. Denying generalities is like denying that we can talk about circles since no object is a perfect circle.